yarn-project/end-to-end/src/e2e_block_building.test.ts

```diff
@@ -51,6 +64,35 @@ describe('e2e_block_building', () => {
       const areDeployed = await Promise.all(receipts.map(r => isContractDeployed(pxe, r.contractAddress!)));
       expect(areDeployed).toEqual(times(TX_COUNT, () => true));
     }, 60_000);
+
+    it('can call public function from different tx in same block', async () => {
+      // Deploy a contract in the first transaction
+      // In the same block, call a public method on the contract
+      const deployer = TokenContract.deploy(owner, owner.getCompleteAddress());
+      await deployer.create();
+
+      const callInteraction = new ContractFunctionInteraction(
+        owner,
+        deployer.completeAddress!.address,
+        TokenContract.abi.functions.find(x => x.name === 'set_minter')!,
+        [minter.getCompleteAddress(), true],
+      );
+
+      await deployer.simulate({});
+      await callInteraction.simulate({
+        // we have to skip simulation of public calls simulation is done on individual transactions
+        // and the tx deploying the contract might go in the same block as this one
+        skipPublicSimulation: true,
+      });
+
+      const receipts = await Promise.all([
+        deployer.send().wait(),
+        callInteraction.send({ skipPublicSimulation: true }).wait(),
+      ]);
+
+      // both transaction should be successful
+      expect(receipts.map(x => x.status)).toEqual(times(2, () => TxStatus.MINED));
```

Comment:
`.wait()` already checks the status to be `MINED`, so you don't need this.

What I would add is the check for both txs to be mined on the same block, given we have seen it working in other tests in this same suite.